prime.go

```diff
@@ -0,0 +1,198 @@
+package merkledag
+
+import (
+	"github.com/ipld/go-ipld-prime"
+	dagpb "github.com/ipld/go-codec-dagpb"
+)
+
+// Kind returns a value from the Kind enum describing what the
+// essential serializable kind of this node is (map, list, integer, etc).
+// Most other handling of a node requires first switching upon the kind.
+func (n *ProtoNode) Kind() ipld.Kind {
```

Comment:
The volume of boilerplate in this file is the main reason I wonder what's the diff between this type and the `go-codec-dagpb.PBNode` type [in my other comment](https://github.com/ipfs/go-merkledag/pull/67/files#r682629402).

It doesn't seem especially harmful to have this if it's providing a useful service, but I don't immediately know what why/what that service is.